p3/plot/_cascade.py

```diff
@@ -376,18 +392,42 @@ def cascade(df, eff=None, **kwargs):
         handles=app_handles, loc="upper left", bbox_to_anchor=(1, 1), ncol=1
     )
 
-    # Attach the platform legend to the bottom of the figure
+    # Attach the platform legend to the position requested by kwargs
     legend_helper = _PlatformLegendHandler(plat_colors, plat_labels)
     plat_handles = [
         mpatches.Patch(color=plat_colors[p], label=p) for p in platforms
     ]
-    fig.legend(
-        handles=plat_handles,
-        loc="upper center",
-        bbox_to_anchor=(0.5, 0.0),
-        handler_map={mpatches.Patch: legend_helper},
-        handlelength=1.0,
-        ncols=4,
-    )
+    if not isinstance(plat_legend_ncols, int):
+        raise ValueError("'plat_legend_ncols' must be an integer")
+    if not isinstance(plat_legend_loc, str):
+        raise ValueError("'plat_legend_loc' must be a string")
+    if plat_legend_loc not in ["north", "east", "south", "west", "off"]:
+        msg = (
+            "'plat_legend_loc' must be 'north', 'east', 'south', 'west', "
+            "or 'off'"
+        )
+        raise ValueError(msg)
+    if plat_legend_loc != "off":
+        if plat_legend_loc == "north":
```

Comment:
Just to make sure: `"north"` = "`lower center"`?

Reply:
Good question.  @swright87, can you explain this?

The code definitely works as expected. My best guess is that the `loc` parameter might be referring to which side of the legend the anchor is attached to... So, when the legend is at the top, the anchor for the legend is at the bottom.

Reply:
Hazy memory but I think I tried logical answers and had unpredictable results, so likely arrived at these values through trial and error (based on the fact the original was plotted on the south with parameter "upper center").

I'm not particularly familiar with matplotlib, generally. [This stack overflow](https://stackoverflow.com/a/44414858) perhaps suggests the parameters are not completely insane.